docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,374 @@
+---
+title: Workload Identity Integration
+authors:
+    - @sonasingh46
+reviewers:
+    - @aramase
+    - @CecileRobertMichon
+
+creation-date: 2022-11-16
+last-updated: N/A
+status: implementable
+see-also:
+    - https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2205
+---
+
+# Workload Identity Integration
+
+## <a name='TableofContents'></a>Table of Contents
+
+<!-- vscode-markdown-toc -->
+* [Table of Contents](#TableofContents)
+* [Acronyms](#Acronyms)
+* [Summary](#Summary)
+* [Motivation](#Motivation)
+	* [Goals](#Goals)
+	* [Non-Goals / Future Work](#Non-GoalsFutureWork)
+* [Proposal](#Proposal)
+	* [Implementation Details/Notes/Constraints](#ImplementationDetailsNotesConstraints)
+		* [Key Generation](#KeyGeneration)
+		* [OIDC URL Setup](#OidcUrlSetup)
+		* [Set Service Account Signing Flags](#SetServiceAccountSigningFlags)
+		* [Federated Credential](#FederatedCredential)
+		* [Distribute Keys To Management Cluster](#DistributeKeys)
+	* [Proposed API Changes](#ProposedApiChanges)
+	* [Proposed Deployment Configuration Changes](#ProposedConfigurationChanges)
+	* [Proposed Controller Changes](#ProposedControllerChanges)
+		* [Identity](#Identity)
+	* [Open Questions](#OpenQuestions)
+		* [1. How to achieve multi-tenancy?](#Howtomultitenancy)
+		* [2. How to distribute key pair to management cluster?](#Howtodistributekeys)
+		* [3. User Experience](#UserExperience)
+	* [Migration Plan](#MigrationPlan)
+	* [Test Plan](#TestPlan)
+* [Implementation History](#ImplementationHistory)
+
+<!-- vscode-markdown-toc-config
+	numbering=false
+	autoSave=false
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+## <a name='Acronyms'></a>Acronyms
+| Acronym      | Full Form               |
+| ------------ | ------------------------|
+| AD           | Active Directory        |
+| AAD          | Azure Active Directory  |
+| AZWI         | Azure Workload Identity |
+| OIDC         | OpenID Connect          |
+| JWKS         | JSON Web Key Sets       |
+
+## <a name='Summary'></a>Summary
+
+Workloads deployed in Kubernetes cluster may require Azure Active Directory application credential or managed identities to access azure protected resource e.g. Azure Key Vault, Virtual Machines etc. Azure AD Pod Identity helps access azure resources without the need of a secret management via Azure Managed Identities. Azure AD Pod Identity is now deprecated and Azure AD Workload Identity is the next iteration of the former. This design proposal aims to define the way for AZWI integration into capz for self managed clusters with keeping in mind other factor e.g. User Experience, Multi Tenancy, Backwards Compatibility etc.
+
```

Comment:
@CecileRobertMichon @aramase -- FWIW, 
I have intentionally written `AZWI integration into capz for self managed clusters` as if I understood correctly for a mangement/bootstrap cluster which is a managed cluster, there is no such special code handling that is required for azwi apart from  meeting the following pre-requisites: 
- Get the OIDC url from managed cluster and steps on how to do that is here https://azure.github.io/azure-workload-identity/docs/installation/managed-clusters.html
- Establish the federated credential using the OIDC url fetched from the above step.  

Yes, it is worth to get this info into the doc and I will put that after y'all see this comment and if it looks ok to you. Missed this part and it came to my mind when I was coding for azwi.